src/zbarlight/__init__.py

```diff
@@ -39,22 +39,34 @@ def scan_codes(code_type, image):
     .. [#zbar_symbologies] http://zbar.sourceforge.net/iphone/userguide/symbologies.html
 
     Args:
-        code_type (str): Code type to search (see ``zbarlight.Symbologies`` for supported values)
+        code_types (list or str): Code type(s) to search (see ``zbarlight.Symbologies`` for supported values)
         image (PIL.Image.Image): Image to scan
 
     returns:
         A list of *code_type* code values or None
 
     """
+    code_types = [code_types] if isinstance(code_types, str) else code_types
+
+    # Translate symbologies
+    symbologies = [
+        Symbologies.get(code_type.upper())
+        for code_type in set(code_types)
+    ]
+
+    # Check that all symbologies are known
+    if None in symbologies:
+        bad_code_types = [code_type for code_type in code_types if code_type not in Symbologies]
```

Comment:
You use `code_type.upper()` as the key in `Symbologies` above when defining `symbologies`, but here you look for `code_type`. As such, the following line could wrongly says that a code type is unknown if it's provided in lower case, if there is also an unknown code type.

Proposal:

    symbologies = []
    for code_type in set(code_types):
        try:
            symbologies.append(Symbologies[code_type.upper()])
        except KeyError:
            raise UnknownSymbologieError('Unknown symbology:  %s' % code_type)

This version raises when it finds the first wrong code type (and does not mention all wrong types as your version does), but I think it's good enough. Alternative:

    symbologies = []
    unknown = []
    for code_type in set(code_types):
        try:
            symbologies.append(Symbologies[code_type.upper()])
        except KeyError:
            unknown.append(code_type)
    if unknown:
        raise UnknownSymbologieError('Unknown symbologies:  %s' % ', '.join(unknown))